library/envirophat/__init__.py

```diff
@@ -1,15 +1,62 @@
 from .i2c_bus import bus
-from .ads1015 import ads1015
-from .bmp280 import bmp280
-from .leds import leds
-from .lsm303d import lsm303d
-from .tcs3472 import tcs3472
+from .ads1015 import ads1015 as _ads1015
+from .bmp280 import bmp280 as _bmp280
+from .leds import leds as _leds
+from .lsm303d import lsm303d as _lsm303d
+from .tcs3472 import tcs3472 as _tcs3472
 
 
 __version__ = '0.0.6'
 
-leds = leds()
-light = tcs3472(bus)
-weather = bmp280(bus)
-analog = ads1015(bus)
-motion = lsm303d(bus)
+
+_is_setup = False
+
+
+class Default(object):
```

Comment:
This is awful. Attempt to defer one-time setup into individual classes. This sort of method call interception can only lead to pain.